inc/admin-interface.php

```diff
@@ -0,0 +1,319 @@
+ *
+ * @since 2.0.0
+ * @package Pantheon_Advanced_Page_Cache
+ */
+
+namespace Pantheon_Advanced_Page_Cache\Admin_Interface;
+
+/**
+ * Kick off the important bits.
+ *
+ * @since 2.0.0
+ * @return void
+ */
+function bootstrap() {
+	// Check if wp_admin_notice exists. We've already noted that the plugin requires at least 6.4, so we're going to not display a notice if you didn't listen to the recommendation.
+	if ( ! function_exists( 'wp_admin_notice' ) ) {
+		add_filter( 'pantheon_apc_disable_admin_notices', '__return_true' );
+	}
+
+	if ( defined( 'PANTHEON_MU_PLUGIN_VERSION' ) ) {
+		// Only do things here if we've got the MU plugin and it's > 1.4.0.
+		if ( version_compare( PANTHEON_MU_PLUGIN_VERSION, '1.4.0', '>' ) ) {
+			// Do stuff, e.g. add_action().
+			add_action( 'admin_notices', __NAMESPACE__ . '\\admin_notice_maybe_recommend_higher_max_age' );
+		} else {
+			add_action( 'admin_notices', __NAMESPACE__ . '\\admin_notice_old_mu_plugin' );
+		}
+	} else {
+		add_action( 'admin_notices', __NAMESPACE__ . '\\admin_notice_no_mu_plugin' );
+	}
+
+	add_filter( 'site_status_tests', __NAMESPACE__ . '\\default_cache_max_age_test' );
+	add_action( 'update_option_pantheon-cache', __NAMESPACE__ . '\\clear_max_age_compare_cache' );
+}
+
+/**
+ * Display an admin notice if the Pantheon MU plugin was not found.
+ *
+ * @since 2.0.0
+ * @return void
+ */
+function admin_notice_no_mu_plugin() {
+	/**
+	 * Allow disabling the admin notice.
+	 *
+	 * @param bool $disable_admin_notices Whether to disable the admin notice.
+	 */
+	if ( apply_filters( 'pantheon_apc_disable_admin_notices', false ) ) {
+		return;
+	}
+
+	wp_admin_notice(
+		// translators: %s is a link to the Pantheon MU plugin.
+		sprintf( __( 'Pantheon Advanced Page Cache works best on the Pantheon platform. If you are working inside a Pantheon environment, ensure your site is using the <a href="%s">Pantheon MU plugin</a>.', 'pantheon-advanced-page-cache' ), 'https://github.com/pantheon-systems/pantheon-mu-plugin' ),
+		[
+			'type' => 'error',
+			'dismissible' => true,
+		]
+	);
+}
+
+/**
+ * Display an admin notice if the Pantheon MU plugin is out of date.
+ *
+ * @since 2.0.0
+ * @return void
+ */
+function admin_notice_old_mu_plugin() {
+	$current_screen = get_current_screen();
+
+	if ( apply_filters( 'pantheon_apc_disable_admin_notices', false ) || 'settings_page_pantheon-cache' !== $current_screen->id ) {
+		return;
+	}
+
+	$mu_plugin_version = PANTHEON_MU_PLUGIN_VERSION;
+	$message = sprintf(
+		// translators: %1$s is a link to the Pantheon MU plugin, %2$s is the version of the MU plugin.
+		__( 'You appear to have an old version of the <a href="%1$s">Pantheon MU plugin</a>. 1.4.0 or above expected but %2$s found.', 'pantheon-advanced-page-cache' ),
+		'https://github.com/pantheon-systems/pantheon-mu-plugin',
+		$mu_plugin_version
+	);
+
+	// Check if there's a composer.json file in the root of the site.
+	if ( file_exists( ABSPATH . 'composer.json' ) ) {
```

Comment:
composer outside web directory isn't supported. nested docroot, yes, because that's what Bedrock does (WP is installed into `/wp`) but the composer file is still in the project root.